### DISCUSSION TITLE
react-router-dom upgrade to v6

### DISCUSSION THREAD

**Discussion by alexanderbartels** [payloadcms/payload#4111] [Feature Requests & Ideas]
Hey, i'm trying to setup payload and remix in a single project/server setup. After installing remix, the /admin route can't be loaded anymore. Browser console shows a webpack error:

```
Uncaught TypeError: _reactRouterDom.withRouter is not a function
```

After some research i found out, that withRouter seems to be deprecated: `Along with the upgrade to v5.1, you should replace any usage of withRouter with hooks.` - https://reactrouter.com/en/main/upgrading/v5#upgrade-to-react-router-v51

Do you have any plans to upgrade react router dom to the latest 6.x version? 

I wanted to avoid the monorepo setup from the example as i'm working on a simple prototype to try out payloadcms and its capabilities.

**Comment by EthanNC** [comment]
Hey @alexanderbartels have you seen [this](https://github.com/manawiki/repay) repo? Might help solve your problem.

**Reply by alexanderbartels** [reply]
thanks for the hint. Have not seen it before. Looks interesting and I will give it a try.

**Reply by j-palindrome** [reply]
This repo is also broken for the same reason—React Router is outdated in Payload and Remix is using v6, so it won't work regardless.

**Comment by ccssmnn** [comment]
For anyone experiencing this bug with npm workspaces:

I had a similar issue using payload in a monorepo with a vite + react frontend. Installing a new react-router-dom@6 package in the frontend made payload receive this one as well. This is an issue with npm workspaces. Dependencies of different versions of a package only get isolated per workspace when the package and version appear in the package.json. react-router-dom doesn't appear in the CMS workspace, as it is a dependency of payload. I was able to fix this by adding `  "peerDependencies": { "react-router-dom": "5.x" },` to the package.json in the workspace with payload. pnpm or yarn have a better dependency isolation story, so I might switch to one of these.
